### PR TITLE
Create authentication

### PR DESCRIPTION
**Pull request by junemichelle**
Authentification with JWT
Move config details to .env file 
Return user details in JSON
Connect to db

### CODE REVIEW

**Comment by FLAWIN** [comment]
Authentication with JWT and moving config to the .env file.

**Comment by Wanguba** [comment]
Returns the user details in JSON format and connects to db

**Comment by VelmaKimani** [comment]
Login returns user details together with the access token.

**Review by junemichelle — COMMENTED** [review]
Don't push the .env file

File: app.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+@app.route('/api/auth/signup', methods=['POST'])
+def signup():
+    data = request.json
+
+    username = data.get('username')
+    email = data.get('email')
+    password = data.get('password')
+
+    if not username or not email or not password:
```
Comment (junemichelle) [inline app.py]: Validate the data passed

File: app.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        'username': username,
+        'password':password,
+        'token': access_token
+                    })
+    set_access_cookies(resp, access_token)
+    return resp, 200
+
+@app.route('/api/protected', methods=['GET'])
+@jwt_required()
+def protected():
```
Comment (junemichelle) [inline app.py]: Explain this

**Comment by ZakariaMoha** [comment]
had unresolved issues

**Comment by ZakariaMoha** [comment]
closed with some issues

**Pull request closed without merging**